src/selectors/per-thread/stack-sample.js

```diff
@@ -267,19 +267,8 @@ export function getStackAndSampleSelectorsPerThread(
   const getTreeOrderComparatorInFilteredThread: Selector<
     (IndexIntoSamplesTable, IndexIntoSamplesTable) => number,
   > = createSelector(
-    threadSelectors.getFilteredThread,
-    getCallNodeInfo,
-    (thread, { callNodeTable, stackIndexToCallNodeIndex }) => {
-      const sampleIndexToCallNodeIndex =
-        ProfileData.getSampleIndexToCallNodeIndex(
-          thread.samples.stack,
-          stackIndexToCallNodeIndex
-        );
-      return ProfileData.getTreeOrderComparator(
-        callNodeTable,
-        sampleIndexToCallNodeIndex
-      );
-    }
+    getSampleIndexToCallNodeIndexForFilteredThread,
+    ProfileData.getTreeOrderComparator
```

Comment:
I see, I'm not using the transforms enough so I don't always think about these use cases.

I don't understand well the case you describe though... Why after collapsing A and C to a new node (let's say A'), the same locations than "previous A and C" are not selected when selecting A'? In other words, why should that always be vertically contiguous in this case?

Reply:
That would be reasonable, too. But it would make drawing more complicated. At the moment, the three states "before selected" / "selected" / "after selected" are enough to draw the graph. If we supported vertically disjointed highlights, we'd need to think of a different way to represent the sample selected states.

Furthermore, I think it might look more noisy visually. For example, let's say you use the JS-only call tree, but your highlights are still in the same position as in the combined call tree, then selecting a JS function might give you "striped" highlight patterns, because the selected function or its ancestors might be running in different JIT tiers which have different call paths in the combined call tree. But this is speculation, it might be fun to try it out!

Oh, and if you focus on a function, samples get filtered out. Would you leave those filtered-out samples in the same spots as in the full thread, too? Then the "filtered out" pattern would need to be drawn in disjointed areas, i.e. both under and over the focused part of the graph. I think the graph could quickly become a jumble of the different patterns (non-highlighted, highlighted, filtered out) if we don't enforce a consistent ordering between them.